### PR TITLE
chore: create pr with client generated with latest master to fabric8-cluster-client

### PR DESCRIPTION
**Pull request by dipak-pawar**
**Changes Proposed in this PR**
* Sourcing bash functions to use from `https://git.io/fxP0E`.
* After successful deployment of latest master, we should expect PR to fabric8-cluster-client repo.

Depends on #openshiftio/openshiftio-cico-jobs/pull/861
Depends on #fabric8-services/fabric8-common/pull/36

Note: We are good to merge if this looks good to you and see are we getting new PR to fabric8-cluster-client repo.

Next Steps:
* [ ] Configure CI jobs to client repo to use new client as per PR and check project is building successfully
* [ ] With merge to master, create a PR with the latest client for its consumer.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -271,6 +271,10 @@ migrate-database: $(BINARY_SERVER_BIN)
 ## Generate GOA sources. Only necessary after clean of if changed `design` folder.
 generate: app/controllers.go migration/sqlbindata.go configuration/confbindata.go
 
+.PHONY: generate-client
+generate-client: $(GOAGEN_BIN)
+	$(GOAGEN_BIN) client -d github.com/fabric8-services/fabric8-cluster/design --pkg cluster
```
Comment (alexeykazakov) [inline Makefile]: Please delete `/cluster` in clean target.

File: cico_setup.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# functions to generate latest client if any change in design pkg
+source <(curl -ssL https://git.io/fxP0E)
```
Comment (alexeykazakov) [inline cico_setup.sh]: Why using your own personal repo?

Comment (dipak-pawar) [reply cico_setup.sh]: no specific reason. I have moved it in fabric8-common. See #https://github.com/fabric8-services/fabric8-common/pull/36

**Comment by sbose78** [comment]
> Configure CI jobs to client repo to use new client as per PR and try to use newly updated client with fabric8-cluster and build the fabric8-cluster project.

@dipak-pawar I understand this step is to verify that the client project builds- but what do we need from `fabric8-cluster` for that?

**Comment by dipak-pawar** [comment]
> I understand this step is to verify that the client project builds- but what do we need from fabric8-cluster for that?

@sbose78 We don't need anything from fabric8-cluster repo. It was little bit confusing. I have updated next steps in desciption.

**Comment by alexeykazakov** [comment]
@dipak-pawar https://github.com/fabric8-services/fabric8-common/pull/36 has been merged. Please update this PR to re-use the scripts from common.

File: cico_setup.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # functions to generate latest client if any change in design pkg
-source <(curl -ssL https://git.io/fxP0E)
+source <(curl -ssL https://git.io/fxD0K)
```
[the remark this answers is not in the document]
Comment (dipak-pawar) [reply cico_setup.sh]: @alexeykazakov done in 6c909ef

File: Makefile (hunk 1 above)
Comment (alexeykazakov) [reply Makefile]: It looks like this change (`/cluster` cleanup) was lost?

Comment (alexeykazakov) [reply Makefile]: My bad. I missed it. Never mind :)

Comment (dipak-pawar) [reply Makefile]: It happens.

**Pull request merged**